unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+                    }
+                } else {
+                    // TODO(egg): Check for unescaped :, @, =, etc. and unescape.
+                }
+                if (isAge) {
+                    return queriedProperty.getSet(
+                            new UnicodePropertySymbolTable.ComparisonMatcher<VersionInfo>(
+                                    UnicodePropertySymbolTable.parseVersionInfoOrMax(propertyValue),
+                                    UnicodePropertySymbolTable.Relation.geq,
+                                    Comparator.nullsFirst(Comparator.naturalOrder()),
+                                    UnicodePropertySymbolTable::parseVersionInfoOrMax));
+                }
+                if (queriedProperty.getName().equals("General_Category")) {
+                    return getGeneralCategorySet(queriedProperties, propertyValue);
+                }
+                return queriedProperty.getSet(propertyValue);
+            }
+        }
+    }
+
+    private static UnicodeSet getIdentitySet(UnicodeProperty queriedProperty) {
+        final var result = new UnicodeSet();
+        for (int cp = 0; cp <= 0x10FFFF; ++cp) {
+            if (UnicodeProperty.equals(cp, queriedProperty.getValue(cp))) {
+                result.add(cp);
+            }
+        }
+        return result;
+    }
+
+    private static UnicodeSet compareProperties(
+            UnicodeProperty queriedProperty, UnicodeProperty comparisonProperty) {
+        if (!((queriedProperty.isType(UnicodeProperty.BINARY_MASK)
+                        && comparisonProperty.isType(UnicodeProperty.BINARY_MASK))
+                || (queriedProperty.isType(UnicodeProperty.NUMERIC_MASK)
+                        && comparisonProperty.isType(UnicodeProperty.NUMERIC_MASK))
+                || (queriedProperty.isType(UnicodeProperty.STRING_MASK)
+                        && comparisonProperty.isType(UnicodeProperty.STRING_MASK))
+                || (queriedProperty.isType(UnicodeProperty.ENUMERATED_OR_CATALOG_MASK)
+                        && comparisonProperty.isType(UnicodeProperty.ENUMERATED_OR_CATALOG_MASK)
+                        && queriedProperty
+                                .getAvailableValues()
+                                .equals(comparisonProperty.getAvailableValues()))
+                || queriedProperty.getName().equals(comparisonProperty.getName()))) {
+            throw new IllegalArgumentException(
+                    "Invalid property comparison between "
+                            + queriedProperty.getTypeName()
+                            + " property "
+                            + queriedProperty.getName()
+                            + " and "
+                            + comparisonProperty.getTypeName()
+                            + " property "
+                            + comparisonProperty.getName());
+        }
+        final var result = new UnicodeSet();
+        for (int cp = 0; cp <= 0x10FFFF; ++cp) {
```

Comment:
Should we note that this excludes empty/multi-character strings of properties-of-strings?

Reply:
Good point.

In my draft spec I express things solely in terms of character properties, and leave properties of strings to extensions.
Including strings seems potentially useful at first glance, but then it rapidly gets messy:
* What is the value of a character property on a string of length other than 1? (null?)
* The set of strings where one property equals another need not be finite (in particular, if two properties are null for all but 1-character strings, that set is the set of all strings of length other than 1).

I’ll add a comment here for now, and think about a note in the draft spec.

Reply:
> * What is the value of a character property on a string of length other than 1? (null?)

We mostly have binary properties of strings, so values are true/yes / false/no.
The value for a string that's not listed is false.

> * The set of strings where one property equals another need not be finite

We discussed this at length. Unicode properties of strings are only defined for finite sets of strings. We define them by writing all strings into a data file. We definitely do not define them via a matching algorithm.

Reply:
> We definitely do not define [properties of strings] via a matching algorithm.

Yes. But in contrast, the value of a property-comparison _is_ defined by matching values.

Reply:
> > We definitely do not define [properties of strings] via a matching algorithm.
> 
> Yes. But in contrast, the value of a property-comparison _is_ defined by matching values.

I think we can compare values on the union of the keySet()'s, plus all code points. That would be well-defined and finite. It's fine if that's a TODO comment or a repo issue for now.